client/entrypoints/express-checkout/index.js

```diff
@@ -479,5 +479,21 @@ jQuery( function ( $ ) {
 		},
 	};
 
-	wcStripeECE.init();
+	// We don't need to initialize ECE on the checkout page now because it will be initialized by updated_checkout event.
+	if (
+		! getExpressCheckoutData( 'is_checkout_page' ) ||
+		getExpressCheckoutData( 'is_pay_for_order' )
+	) {
```

Comment:
```suggestion
	if ( ! getExpressCheckoutData( 'is_checkout_page' ) ) {
```
This condition looks weird. The second part is redundant; if it's the pay-for-order page, then the first condition would've been true.

Or maybe we want to initialize ECE just for the pay-for-order? (with `updated_cart_totals` `updated_checkout` initializing it in the other pages 🤷🏼

Reply:
You are right. Only the first condition (checking if it is the checkout page) is needed to avoid double `init` on the checkout page. We have to call the ' init ' for all other pages (product, cart, pay for order).

I have fixed it in [3b83dc3](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3490/commits/3b83dc3401f5fb44fc6c135c81535c79a1afa4c9)